lib/cretonne/meta/cdsl/ast.py

```diff
@@ -88,6 +90,40 @@ def copy(self, m):
 
         return Def(tuple(new_defs), new_expr)
 
+    def definitions(self):
+        # type: () -> Set[Var]
+        """ Return the set of all Vars that are defined by self"""
+        return set(self.defs)
+
+    def uses(self):
+        # type: () -> Set[Var]
+        """ Return the set of all Vars that are used(read) by self"""
+        return set(self.expr.vars())
+
+    def vars(self):
+        # type: () -> Set[Var]
+        """ Return the set of all Vars that appear in self"""
+        return self.definitions().union(self.uses())
+
+    def substitution(self, other, s):
+        # type: (Def, VarMap) -> Optional[VarMap]
+        """
+        If the Defs self and other agree structurally, return a variable
+        substitution to transform self ot other. Two Defs agree structurally
```

Comment:
typo: ot

Also, could you clarify in the documentation what it means to "agree structurally"?